src/client/render/pipeline.cpp

```diff
@@ -225,10 +225,12 @@ video::ITexture *DynamicSource::getTexture(u8 index)
 
 void ScreenTarget::activate(PipelineContext &context)
 {
-	auto driver = context.device->getVideoDriver();
-	driver->setRenderTarget(nullptr, m_clear, m_clear, context.clear_color);
-	driver->OnResize(size);
-	RenderTarget::activate(context);
+	if (!context.client->getRenderingEngine()->headless) {
```

Comment:
Less likely to have merge conflicts with upstream if you do an early return, also I think a bit more readable.
`if (context.client->getRenderingEngine()->headless) return;`